src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -262,12 +262,12 @@ public Guid CallSiteID
         #region public methods
 
         /// <summary>
-        /// Load the serialised data provided into this callsite's trace cache
+        /// Load the serialized data provided into this callsite's trace cache
         /// </summary>
         /// <param name="serializedTraceData">The data to load</param>
         public void LoadSerializedDataIntoTraceCache(string serializedTraceData)
         {
-            if (serializedTraceData == null)
+            if (serializedTraceData == null || CheckIfTraceDataIsLegacySOAPFormat(serializedTraceData))
```

Comment:
What happens if you skip the `CheckIfTraceDataIsLegacy...`?

Reply:
then the try catch below will get hit and we'll end up doing a full base64 decode and catching an exception.

Reply:
The thing we can do regarding your question about earlier catches and optimization of this check is handle something in here -> https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Workspaces/SerializationConverters.cs.  This is the only place dynamo core encounters the Soap data.  Maybe you can catch the old data there and then here the input string would be `null`.  That might also help you with your task @aparajit-pratap .  The nice thing is that the SerializationConverter is doing the whole `Bindings` section of the graph at once vs here were it is per `callsite`

Reply:
In the D4C3D case we may just need to communicate the same pattern as the load data from dwg file.

Reply:
This optimization can be done later too.  This function could stay the same.

Reply:
Actually even better, if you catch bad data at the file deserialization then even the `GetOrphanedSerializablesAndClearHistoricalTraceData` pass would be short circuited.  It is looking at `HomeWorkSpace.PreloadedTraceData` which is set during deserialization.

Reply:
Maybe it's just as simple as checking the DynamoVersion?   Not full proof but in theory you should never have XML graph or a 2.x graph with good data.